src/runtime/tests/manifest-test.ts

```diff
@@ -3659,6 +3659,130 @@ resource SomeName
+          particle Foo
+            data: reads {*}
+      `), `\
+Post-parse processing error caused by 'undefined' line 3.
+Only type variables may have '*' fields.
+              data: reads {*}
+                          ^^^`);
+      await assertThrowsAsync(async () => await parseManifest(`
+          particle Foo
+            data: reads {name: Text, *}
+      `), `\
+Post-parse processing error caused by 'undefined' line 3.
+Only type variables may have '*' fields.
+              data: reads {name: Text, *}
+                          ^^^^^^^^^^^^^^^`);
+    });
+
+    it('warns about using multiple `*` in a single variable constraint', async () => {
+      const manifest = await parseManifest(`
+          particle Foo
+            data: reads ~a with {*, *}
+      `);
+
+      assert.lengthOf(manifest.errors, 1);
+      assert.equal(manifest.errors[0].key, 'multiStarFields');
+    });
+
+    it('supports field-level checks and claims with max type variables', async () => {
+      const manifest = await parseManifest(`
+        particle OrderIngestion in '.OrderIngestion'
+          data: writes [Product {sku: Text, name: Text, price: Number}]
+
+        particle SkuRedactor in '.SkuRedactor'
+          input: reads [~a with {sku: Text}]
+          output: writes [~a]
+          claim output.sku is redacted
+
+        particle Egress in '.Egress'
+          data: reads [~x with {sku: Text, *}]
+          check data.sku is redacted
+
+        recipe Shop
+          beforeRedaction: create
+          afterRedaction: create
+          OrderIngestion
+            data: beforeRedaction
+          SkuRedactor
+            input: beforeRedaction
+            output: afterRedaction
+          Egress 
+            data: afterRedaction
+      `);
+      const recipe = manifest.recipes[0];
+
+      // Normalize and clone the recipe. This resolves the type variables, and
+      // exercises code paths to validate the checks and claims.
+      // Should not crash.
+      recipe.normalize();
+      const clonedRecipe = recipe.clone();
+
+      // Verify claim on redactor particle.
+      const redactorParticle = clonedRecipe.particles.find(p => p.name === 'SkuRedactor').spec;
+      const outputType = redactorParticle.connections.find(c => c.name === 'output').type as TypeVariable;
+      outputType.maybeEnsureResolved();
+      assert.isTrue(outputType.isResolved());
+      assert.lengthOf(redactorParticle.trustClaims, 1);
+      const claim = redactorParticle.trustClaims[0];
+      assert.deepStrictEqual(claim.fieldPath, ['sku']);
+
+      // Verify claim and type on egress particle.
+      const egressParticle = clonedRecipe.particles.find(p => p.name === 'Egress').spec;
+      assert.lengthOf(egressParticle.trustChecks, 1);
+      const check = egressParticle.trustChecks[0];
+      assert.deepStrictEqual(check.fieldPath, ['sku']);
+    });
+
+    it('resolves to max type through an intermediary', async () => {
```

Comment:
Is this the same as the test in type-checker-test.ts  ? If almost the same then you can remove this one.

Reply:
Given your next comment, I'm keeping this test and marking this as resolved.

Reply:
I understand better what you mean given chat. Will delete this method.